openreview/journal/invitation.py

```diff
@@ -6433,6 +6434,69 @@ def set_note_reviewer_message_invitation(self, note):
             signatures=[self.journal.venue_id]
         )
 
+    def set_preferred_emails_invitation(self):
+
+        venue_id = self.journal.venue_id
+
+        if openreview.tools.get_invitation(self.client, self.journal.get_preferred_emails_invitation_id()):
+            return
+
+        invitation = Invitation(
+            id=self.journal.get_preferred_emails_invitation_id(),
+            invitees=[venue_id],
+            readers=[venue_id],
+            writers=[venue_id],
+            signatures=['~Super_User1'], ## it should be the super user to get full email addresses
+            minReplies=1,
+            maxReplies=1,
+            type='Edge',
+            edit={
+                'id': {
+                    'param': {
+                        'withInvitation': self.journal.get_preferred_emails_invitation_id(),
+                        'optional': True
+                    }
+                },                
+                'ddate': {
+                    'param': {
+                        'range': [ 0, 9999999999999 ],
+                        'optional': True,
+                        'deletable': True
+                    }
+                },
+                'cdate': {
+                    'param': {
+                        'range': [ 0, 9999999999999 ],
+                        'optional': True,
+                        'deletable': True
+                    }
+                },                
+                'readers': [venue_id, self.journal.get_action_editors_id(), '${2/head}'],
```

Comment:
What is `preferred_emails_groups` for? I thought that variable was going to appear here.

Reply:
it is a setting to indicate which groups we are going to save their preferred email address. This setting is being used in the date process function that computes the edges: https://github.com/openreview/openreview-py/pull/2196/files#diff-3cb41621f071b32bebe35fc3d382c965d8ebf5a92c3aadb9ea4c18d732868929